app/components/theme/mobile_switcher_component.html.erb

```diff
@@ -0,0 +1,28 @@
+<div data-controller="visibility" data-action="visibility:click:outside->visibility#off" class="flex items-center space-x-2 w-full">
+  <button type="button" data-action="click->visibility#toggle" data-test-id="sort-select" class="relative w-full cursor-default rounded-md bg-white dark:bg-gray-700 py-1.5 pl-3 pr-10 text-left text-gray-900 dark:text-gray-200 shadow-sm ring-1 ring-inset ring-gray-300 dark:ring-gray-600 focus:outline-none focus:ring-2 focus:ring-gray-600 dark:focus:ring-gray-500 sm:text-sm sm:leading-6" aria-haspopup="listbox" aria-expanded="true" aria-labelledby="listbox-label">
+    <span class="text-gray-600 hover:bg-gray-50 hover:text-gray-900 px-1 py-1 text-base font-medium dark:text-gray-300 dark:hover:text-gray-200 dark:hover:bg-gray-700/60 "><%= current_theme.name.capitalize %> mode</span>
+    <span class="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-2">
+      <%= inline_svg_tag 'icons/arrows-up-down.svg', class: 'h-5 w-5 text-gray-400 dark:text-gray-300', aria: true, title: 'Select Theme' %>
+    </span>
+  </button>
+
+  <ul
+    data-visibility-target="content"
+    data-transition-enter=""
+    data-transition-enter-start=""
+    data-transition-enter-end=""
+    data-transition-leave="transition ease-in duration-100"
+    data-transition-leave-start="opacity-100"
+    data-transition-leave-end="opacity-0"
+    class="absolute hidden z-10 mt-1 max-h-60 w-full min-w-max rounded-md bg-white dark:bg-gray-700 py-1 text-base shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none sm:text-sm" tabindex="-1" role="listbox" aria-labelledby="listbox-label" aria-activedescendant="listbox-option-3">
+
+      <% Users::Theme.default_themes.each do |theme| %>
+        <%= link_to themes_path(theme: theme.name), class: 'text-gray-700 dark:text-gray-300 group flex items-center px-3 py-2 text-sm', data: { turbo_method: :put, controller: 'theme-switcher', 'theme-switcher-theme-value': current_theme.to_s } do %>
```

Comment:
Having the controller on all of these links will create 3 different instances of the ThemeController that will each call connect, we'll be updating the root element 3 times on every page request.

We'd be best to only have one instance of the ThemeController on the parent ul element and change the options to li elements instead of links. Then, when an option is clicked, it should trigger a function on the controller like the update theme.

That means we'd also have to move the turbo request for updating the theme in the backend to within the controller. But, that shouldn't be difficult. We do something very similar in the [SharePreviewController](https://github.com/TheOdinProject/theodinproject/blob/main/app/javascript/controllers/share_preview_controller.js#L17) you can use a reference.